clang/runtime/dpct-rt/include/memory.hpp.inc

```diff
@@ -1906,6 +1906,63 @@ using constant_memory = detail::device_memory<T, constant, Dimension>;
 template <class T, size_t Dimension>
 using shared_memory = detail::device_memory<T, shared, Dimension>;
 // DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|dpct_allocator|dpct
+// DPCT_DEPENDENCY_BEGIN
+// Device|get_default_queue
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+template <typename T, sycl::usm::alloc AllocKind = sycl::usm::alloc::host>
+class allocator {
```

Comment:
Naming the wrapper class just `allocator` seems too generic to me. Maybe we can be more specific like the mapping target `pinned_allocator`?

Reply:
> Naming the wrapper class just `allocator` seems too generic to me. Maybe we can be more specific like the mapping target `pinned_allocator`?

What about naming the wrapper to `host_allocator` ? At the same time, we remove the second template argument and use `sycl::usm::alloc::host` as fixed allocation type inside the wrapper.

`sycl::usm_allocator<T, sycl::usm::alloc::host>` seems to be a close mapping for `pinned_allocator<T>`. Both allocate/deallocate memory in host side, except the gap there is no "pinned memory" in SYCL.

Reply:
@reble
I uploaded da9515dbaf933dfdcf5b9bcde34978ea3dd8c0e6 to refine the wrapper class name. Please help review.
- Moved `dpct::allocator` to `dpct::detail::usm_allocator`
- Added `dpct::usm_host_allocator<T>` alias to `dpct::detail::usm_allocator<T, sycl::usm::alloc::host>`

Below is an migration example:
CUDA:
`std::vector<float, thrust::cuda::experimental::pinned_allocator<float>> hVecCopy = hVec;`
Migrated SYCL:
`std::vector<float, dpct::usm_host_allocator<float>> hVecCopy = hVec`